### PR TITLE
fix: ServerSentEvent typing error

### PR DESCRIPTION
**Pull request by euri10**
fixes small typing error:

```
error: Argument 1 to "ServerSentEvent" has incompatible type "AsyncIterable[ServerSentEventMessage]"; expected "str | bytes | Iterable[str | bytes] | Iterator[str | bytes] | AsyncIterable[str | bytes] | AsyncIterator[str | bytes]"  [arg-type]
 ```

inside `test_sse` there was a `Any` I changed to trigger the test then solved it.

### CODE REVIEW

**Review by provinzkraut — CHANGES_REQUESTED** [review]

File: litestar/types/helper_types.py [hunk 1]
```diff
@@ -15,9 +17,12 @@
     Union,
 )
 
+from litestar.response.sse import ServerSentEventMessage
```
Comment (euri10) [inline litestar/types/helper_types.py:20]: CI fails for a weird reason to me, mentionning circular imports (https://github.com/litestar-org/litestar/actions/runs/7713504937/job/21023551864?pr=3048) and this is the only instance  can think in the PR could cause this, however at first  put him under TPYE_CHECKING but ruff was complaining

Comment (euri10) [reply litestar/types/helper_types.py:20]: if I put it under TPYE_CHECKING I get:
litestar/types/helper_types.py:23:39: TCH004 Move import `litestar.response.sse.ServerSentEventMessage` out of type-checking block. Import is used for more than type hinting.

Comment (provinzkraut) [reply litestar/types/helper_types.py:20]: > Import is used for more than type hinting.

You'll need to put the import in quotes. I left a comment down below!

File: litestar/types/helper_types.py [hunk 2]
```diff
@@ -37,3 +42,6 @@
 
 MaybePartial: TypeAlias = Union[T, partial]
 """A potentially partial callable."""
+
+SSEData: TypeAlias = Union[int, str, bytes, Dict[str, Any], ServerSentEventMessage]
```
Comment (provinzkraut) [inline litestar/types/helper_types.py:46]:
```suggestion
SSEData: TypeAlias = Union[int, str, bytes, Dict[str, Any], "ServerSentEventMessage"]
```

File: litestar/types/helper_types.py [hunk 3]
```diff
@@ -15,9 +17,12 @@
     Union,
 )
 
+from litestar.response.sse import ServerSentEventMessage
+
 if TYPE_CHECKING:
     from typing_extensions import TypeAlias
 
+
```
Comment (provinzkraut) [inline litestar/types/helper_types.py:25]:
```suggestion

if TYPE_CHECKING:
    from typing_extensions import TypeAlias
    from litestar.response.sse import ServerSentEventMessage


```

Comment (euri10) [reply litestar/types/helper_types.py]: done, thanks for bearing with me

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**